test/sol6/daoFuzzTests.js

```diff
@@ -0,0 +1,434 @@
+const TestToken = artifacts.require('TestToken.sol')
+const DAOContract = artifacts.require('MockKyberDaoMoreGetters.sol')
+// using mock contract here, as we need to read the hasInited value
+const KyberStaking = artifacts.require('KyberStaking.sol')
+
+const Helper = require('../helper.js')
+const {precisionUnits, zeroBN, zeroAddress, BPS} = require('../helper.js')
+const BN = web3.utils.BN
+const {expectRevert} = require('@openzeppelin/test-helpers')
+const {DEPOSIT, DELEGATE, WITHDRAW, NO_ACTION} = require('../../scripts/simulator/stakingActionsGenerator.js')
+const {
+  CREATE_CAMPAIGN,
+  CANCEL_CAMPAIGN,
+  GET_REWARD,
+  VOTE,
+  CAMPAIGN_TYPE_GENERAL,
+  CAMPAIGN_TYPE_NETWORK_FEE,
+  CAMPAIGN_TYPE_FEE_BRR
+} = require('../../scripts/simulator/daoActionsGenerator.js')
+
+const DaoSimulator = require('../../scripts/simulator/simulator_dao.js')
+const StakeGenerator = require('../../scripts/simulator/stakingActionsGenerator.js')
+const DaoGenerator = require('../../scripts/simulator/daoActionsGenerator.js')
+
+let kncToken
+let tokenDecimals = new BN(18)
+let admin
+let daoSetter
+let stakingContract
+let feeHandler
+let daoContract
+let totalKNC
+let currentBlockTime
+
+const POWER_128 = new BN(2).pow(new BN(128))
+
+//dao-related variable
+let startTime
+let epochPeriod = 500 // each iteration we add 10s -> each epoch is about 50 iterations
+let minCampPeriod = 0
+let latestNetworkFee = 25
+let latestRewardBps = new BN(3000) // 30%
+let latestRebateBps = new BN(2000) // 20%
+let link = web3.utils.fromAscii('https://kyberswap.com')
+
+let NUM_RUNS = 250
+// let NUM_RUNS = 5000
+
+// statistic about operation
+// for operation, failing means the generated operation is revert
+// for campaign, failing means the winning option is 0
+let score = {
+  deposit: {success: 0, fail: 0},
+  delegate: {success: 0, fail: 0},
+  withdraw: {success: 0, fail: 0},
+  submitNewCampaign: {success: 0, fail: 0},
+  cancelCampaign: {success: 0, fail: 0},
+  vote: {success: 0, fail: 0},
+  noAction: {success: 0, fail: 0},
+  successCampaign: {success: 0, fail: 0}
+}
+
+contract('KyberDAO simulator', function (accounts) {
```

Comment:
lets call it fuzzing kyberDao or maybe kyberDao fuzzer

Simulator is only for validating results. its not the essence of the test.